src/test/url-handling.test.js

```diff
@@ -473,6 +474,120 @@ describe('showTabOnly', function() {
   });
 });
 
+describe('ranges', function() {
+  describe('are serializing correctly', () => {
+    it('when there is no range', () => {
+      const { getState } = _getStoreWithURL();
+      const urlState = urlStateReducers.getUrlState(getState());
+      const queryString = getQueryStringFromUrlState(urlState);
+      expect(queryString).not.toContain(`range=`);
+    });
+
+    it('when there is 1 range', () => {
+      const { getState, dispatch } = _getStoreWithURL();
+
+      dispatch(commitRange(1514.587845, 25300));
+      const urlState = urlStateReducers.getUrlState(getState());
+      const queryString = getQueryStringFromUrlState(urlState);
+      expect(queryString).toContain(`range=1514m23786`); // 1.514s + 23786ms
+    });
+
+    it('when rounding down the start', () => {
+      const { getState, dispatch } = _getStoreWithURL();
+
+      dispatch(commitRange(1510.58, 1519.59));
+      const urlState = urlStateReducers.getUrlState(getState());
+      const queryString = getQueryStringFromUrlState(urlState);
+      expect(queryString).toContain(`range=1510m10`); // 1.510s + 10ms
+    });
+
+    it('when the duration is 0', () => {
+      const { getState, dispatch } = _getStoreWithURL();
+
+      dispatch(commitRange(1514, 1514));
+      const urlState = urlStateReducers.getUrlState(getState());
+      const queryString = getQueryStringFromUrlState(urlState);
+      expect(queryString).toMatch(/range=1514000000n(?!0)/); // 1.514s + something non zero
+    });
+
+    it('when there are several ranges', () => {
+      const { getState, dispatch } = _getStoreWithURL();
+
+      dispatch(commitRange(1514.587845, 25300));
+      dispatch(commitRange(1800, 1800.1));
```

Comment:
I believe this would be something like `300000000n10` for a 5min - 5min+10 ns range. (5min is 300 seconds) I think this isn't too bad because this is only for a few users and this won't likely be present a lot (by construction, small ranges are at the end of the list of decreasing ranges).